python/GUARDDUTY_UNTREATED_FINDINGS/GUARDDUTY_UNTREATED_FINDINGS.py

```diff
@@ -0,0 +1,509 @@
+    guardduty_detector_status = guardduty_client.get_detector(
+        DetectorId=guardduty_detector_id
+    )
+    if guardduty_detector_status['Status'] == 'DISABLED':
+        return None
+
+    # get all the guardduty findings
+    all_findings_list = list_all_findings(guardduty_client, guardduty_detector_id)
+
+    # If no findings present then return COMPLIANT
+    if not all_findings_list:
+        return build_evaluation(event['accountId'], 'COMPLIANT', event)
+
+    # evaluate each finding
+    for each_finding in all_findings_list:
+        days_since_created = get_delta_days(each_finding['CreatedAt'])
+
+        # 0.1 to 3.9 severity score is LowSev
+        if 0.1 <= each_finding['Severity'] <= 3.9:
+            if days_since_created <= valid_rule_parameters['daysLowSev']:
+                evaluations.append(build_evaluation(each_finding['Id'], 'COMPLIANT', event))
+                continue
+            evaluations.append(build_evaluation(each_finding['Id'], 'NON_COMPLIANT', event, annotation='This AWS GurdDuty Low Severity finding is older than {} days'.format(valid_rule_parameters['daysLowSev'])))
+            continue
+
+        # 4.0 to 6.9 severity score is MediumSev
+        if 4.0 <= each_finding['Severity'] <= 6.9:
+            if days_since_created <= valid_rule_parameters['daysMediumSev']:
+                evaluations.append(build_evaluation(each_finding['Id'], 'COMPLIANT', event))
+                continue
+            evaluations.append(build_evaluation(each_finding['Id'], 'NON_COMPLIANT', event, annotation='This AWS GurdDuty Medium Severity finding is older than {} days'.format(valid_rule_parameters['daysMediumSev'])))
+            continue
+
+        # 7.0 to 8.9 severity score is HighSev
+        if 7.0 <= each_finding['Severity'] <= 8.9:
+            if days_since_created <= valid_rule_parameters['daysHighSev']:
+                evaluations.append(build_evaluation(each_finding['Id'], 'COMPLIANT', event))
+                continue
+            evaluations.append(build_evaluation(each_finding['Id'], 'NON_COMPLIANT', event, annotation='This AWS GurdDuty High Severity finding is older than {} days'.format(valid_rule_parameters['daysHighSev'])))
+            continue
+
+    return evaluations
+
+
+# get number of days since the finding is created
+def get_delta_days(finding_creation_time):
+    current_datetime_object = datetime.utcnow()
+    creation_datetime_object = datetime.strptime(finding_creation_time, '%Y-%m-%dT%H:%M:%S.%fZ')
+    delta_time = current_datetime_object - creation_datetime_object
+    return delta_time.days
+
+# function to get all findings. Each call supports maximum of 50 items.
+def list_all_findings(guardduty_client, guardduty_detector_id):
+    all_findings_list = []
+    findings_id_list = guardduty_client.list_findings(DetectorId=guardduty_detector_id)
+    if not findings_id_list['FindingIds']:
+        return []

```

Comment:
Can we return all_findings_list here as it will be a empty list initially and it will be equivalent to returning []

Reply:
Modified it to return `all_findings_list`